src/components/survey/sorting-hat.tsx

```diff
@@ -0,0 +1,255 @@
+  answers: {},
+}
+
+// const SORTING_HAT_KEY = `egghead_sorting_hat`
+
+const cioIdentify = (id: string, answers: any) => {
+  if (id) {
+    window._cio.identify({
+      id,
+      last_surveyed_at: Math.round(Date.now() / 1000),
+      ...answers,
+    })
+  }
+}
+
+const sortingHatReducer = (state: any, action: any) => {
+  const question: any = sortingHatData[state.currentQuestion]
+  const now = Math.round(Date.now() / 1000)
+
+  const getSavedState = () => {
+    const savedState = false //localStorage.getItem(SORTING_HAT_KEY)
+    if (savedState) {
+      return JSON.parse(savedState)
+    } else {
+      return false
+    }
+  }
+  const findCurrentQuestion = (question: string, answers: any): string => {
+    if (question && answers[question]) {
+      const possibleNext = sortingHatData[question].next?.[answers[question]]
+      if (![answers[possibleNext]]) {
+        return possibleNext
+      } else {
+        return findCurrentQuestion(possibleNext, answers)
+      }
+    } else {
+      return question || 'thanks'
+    }
+  }
+  switch (action.type) {
+    case `load`:
+      const savedState = getSavedState() || state
+
+      if (action.subscriber) {
+        let attributes = get(action.subscriber, 'attributes', {})
+        cioIdentify(action.subscriber.id, savedState.answers)
+        const answers = {
+          ...savedState.answers,
+          ...action.subscriber.attributes,
+        }
+
+        if (isEmpty(attributes.sorting_hat_started_at)) {
+          cioIdentify(action.subscriber.id, {
+            sorting_hat_started_at: now,
+          })
+          track(`started survey`, {
+            survey: 'sorting hat',
+          })
+          attributes = {...attributes, sorting_hat_started_at: now}
+        }
+
+        if (isEmpty(attributes.sorting_hat_finished_at)) {
+          return {
+            ...state,
+            ...savedState,
+            answers,
+            subscriber: {
+              ...action.subscriber,
+              attributes,
+            },
+            currentQuestion: findCurrentQuestion(
+              savedState.currentQuestion,
+              answers,
+            ),
+            loading: false,
+          }
+        } else {
+          return {closed: true}
+        }
+      } else if (action.loadingSubscriber === false) {
+        return {
+          closed: true,
+          error: 'You must be logged in to take the survey!',
+        }
+      } else {
+        return state
+      }
+    case `answered`:
+      const isFinal = question.final
+
+      const answers = {
+        ...state.answers,
+        ...state.subscriber.attributes,
+        [state.currentQuestion]: action.answer,
+      }
+
+      const nextQuestion = findCurrentQuestion(
+        question.next[action.answer],
+        answers,
+      )
+
+      const updatedState = {...state, answers, currentQuestion: nextQuestion}
+
+      // localStorage.setItem(SORTING_HAT_KEY, JSON.stringify(updatedState))
```

Comment:
I was fiddling with a localstorage option that would allow us to survey without requiring a CIO subscriber but bailed on it for now